cmd/argo/commands/submit.go

```diff
@@ -81,38 +77,26 @@ func SubmitWorkflows(filePaths []string, submitOpts *util.SubmitOpts, cliOpts *c
 		cliOpts = &cliSubmitOpts{}
 	}
 	defaultWFClient := InitWorkflowClient()
-	var workflows []wfv1.Workflow
+
+	var fileContents [][]byte
 	if len(filePaths) == 1 && filePaths[0] == "-" {
-		reader := bufio.NewReader(os.Stdin)
-		body, err := ioutil.ReadAll(reader)
+		body, err := util.ReadFromStdin()
```

Comment:
We can also create another method which contains `ReadFromStdin` and `ReadFromFilePathsOrUrls` like `ReadFromStdinOrFilePathsOrUrls`. Please consider another name for the function as `ReadFromStdinOrFilePathsOrUrls` is too long.

Reply:
I also thought about doing the same, but I stopped because:
- it would break the single responsibility principle (it already is in ```ReadFromFilePathsOrUrls``` but I thought of it as a necessary evil ) and make the function harder to test
- The name would be too long and I couldn't think of another name.

Reply:
> The name would be too long and I couldn't think of another name.

How about `ReadManifests`?

> it would break the single responsibility principle (it already is in ReadFromFilePathsOrUrls but I thought of it as a necessary evil ) and make the function harder to test

I see. If there's a good reason, it's totally fine!

Reply:
I can see that adding a ```ReadManifests``` function would make it easier to add code for another ```submit``` or ```create``` command such as ```argo cron submit```.

I will add it.